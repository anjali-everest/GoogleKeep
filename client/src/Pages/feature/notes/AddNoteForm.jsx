import React, { useState } from 'react'
import { useDispatch } from "react-redux"
import { addNewNote } from "./ThunkHandler"
import ResizableTextarea from '../../ResizableTextArea'

export const AddNoteForm = () => {
  const dispatch = useDispatch()
  const [openForm, setOpenForm] = useState(false)
  const [title, setTitle] = useState('')
  const [content, setContent] = useState('')

  const onContentClicked = e => setOpenForm(true)
  const onTitleChanged = e => setTitle(e.target.value)
  const onContentChanged = e => setContent(e.target.value)
  const onSaveNoteClicked = () => {
    if (title && content) {
      dispatch(
        addNewNote({
          title,
          content
        })
      )

      setTitle('')
      setContent('')
    }
  }

  return (
    <div className="add-note-form">
      <input
        type="text"
        id="noteTitle"
        name="noteTitle"
        className="note-title"
        hidden={!openForm}
        value={title}
        placeholder="Title"
        onChange={onTitleChanged}
      />
      <ResizableTextarea placeholder="Take a note..."
        value={content}
        onChange={onContentChanged}
        onClick={onContentClicked}
      />
      <button type="button" hidden={!openForm} className="add-note-button" onClick={onSaveNoteClicked}>Save</button>
    </div>
  )
}