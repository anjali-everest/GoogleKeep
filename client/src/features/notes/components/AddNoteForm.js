import React, { useState } from "react";
import { useDispatch } from "react-redux";
import { addNewNote } from "../thunkHandler";
import ResizableTextarea from "../../../components/ResizableTextArea";
import { NOTE_TITLE_PLACEHOLDER } from "../../../util/AppConstants";

export const AddNoteForm = () => {
  const dispatch = useDispatch();
  const [openForm, setOpenForm] = useState(false);
  const [title, setTitle] = useState("");
  const [content, setContent] = useState("");

  const onContentClicked = () => setOpenForm(true);
  const onTitleChanged = (e) => setTitle(e.target.value);
  const onContentChanged = (e) => setContent(e.target.value);
  const onSaveNoteClicked = () => {
    if (title || content) {
      dispatch(
        addNewNote({
          title,
          content,
        })
      );

      setTitle("");
      setContent("");
    }
  };

  return (
    <div className="add-note-form">
      <input
        type="text"
        id="noteTitle"
        name="noteTitle"
        className="noteTitle"
        hidden={!openForm}
        value={title}
        placeholder={NOTE_TITLE_PLACEHOLDER}
        onChange={onTitleChanged}
      />
      <ResizableTextarea
        id="noteContent"
        placeholder={NOTE_TITLE_PLACEHOLDER}
        value={content}
        onChange={onContentChanged}
        onClick={onContentClicked}
      />
      <button
        type="button"
        hidden={!openForm}
        className="add-note-button"
        onClick={onSaveNoteClicked}
      >
        Save
      </button>
    </div>
  );
};
