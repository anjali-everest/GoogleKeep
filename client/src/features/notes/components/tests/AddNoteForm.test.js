import React from "react";
import { mount } from "enzyme";
import { AddNoteForm } from "../AddNoteForm";
import configureMockStore from "redux-mock-store";
import * as ReactReduxHooks from "react-redux";

const mockStore = configureMockStore();

describe("AddNoteForm tests", () => {
  const store = mockStore({
    notes: [],
    status: "idle",
    error: null,
    editingNoteId: null,
    isEditingNote: false,
    deletingNoteId: null,
  });

  beforeEach(() => {
    jest
      .spyOn(ReactReduxHooks, "useSelector")
      .mockImplementation(() => store.getState());
    jest
      .spyOn(ReactReduxHooks, "useDispatch")
      .mockImplementation(() => store.dispatch);
  });

  it("Should render without crashing", () => {
    mount(<AddNoteForm />);
  });

  it("should match the AddNoteForm snapshot", function () {
    const addNoteForm = mount(<AddNoteForm />);
    expect(addNoteForm).toMatchSnapshot();
  });
});
