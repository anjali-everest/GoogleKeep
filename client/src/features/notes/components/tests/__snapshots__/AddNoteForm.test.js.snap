// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNoteForm tests should match the AddNoteForm snapshot 1`] = `ReactWrapper {}`;
